docs/user-guide/deploying-application/image-labels-and-comments.md

```diff
@@ -0,0 +1,115 @@
+
+### From Build History
+
+![Figure 3: Adding Labels and Comments - 'Build History' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-build-history.gif)
+
+### From Deployment History
+
+![Figure 4: Adding Labels and Comments - 'Deployment History' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-deployment-history.gif)
+
+### From App Details
+
+![Figure 5: Adding Labels and Comments - 'App Details' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-app-details.gif)
+
+---
+
+## Deleting Labels & Comments
+
+### Soft-Delete Labels
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have [Build & deploy permission](../global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to perform soft deletion of labels.
+{% endhint %}
+
+This action marks the label as invalid but doesn't delete the label. Therefore, you can recover it again.
+
+1. Click the edit option.
+2. Use the (-) icon to strike off the label. This icon is available on the left-side of a label.
+3. Click **Save**. 
+
+![Figure 6: Soft Deletion of a Label](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/soft-delete-tag.gif)
+
+### Hard-Delete Labels
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have super-admin permission to perform hard deletion of labels.
+{% endhint %}
+
+This action deletes the label and makes it available for reuse in same or any other image.
+
+1. Click the edit option.
+2. Use the (x) icon to permanently remove the label. This icon is available on the right-side of a label.
+3. Click **Save**.
+
+![Figure 7: Hard Deletion of a Label](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/hard-delete-tag.gif)
+
+### Removing Comments
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have [Build & deploy permission](../global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to remove comments.
+{% endhint %}
+
+If you wish to permanently remove a comment, do the following:
+
+1. Click the edit option.
+2. Empty the content of an existing comment.
+3. Click **Save**.
+
+![Figure 8: Removing a Comment](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/remove-comment.gif)
+
+---
+
+## Extra Use Case
+
+If you use [Application Groups](../application-groups.md) to deploy in bulk, image labels (if added) will be available as filters for you to quickly locate the container image.
```

Comment:
Please mention this in Application Group page as well, with a hyperlink to this page.

Reply:
Also please make a note of mentioning this in the Release documentation as well

Reply:
Done

Reply:
Added the second in Masterlist's ToDo sheet